drivers/hwinfo/Kconfig

```diff
@@ -10,7 +10,10 @@ menuconfig HWINFO
 
 if HWINFO
 
-config HWINFO_HAS_DRIVER
+config HWINFO_DEVICE_ID_HAS_DRIVER
+	bool
+
+config HWINFO_RESET_CAUSE_HAS_DRIVER
```

Comment:
We have weak implementations so there is no good reason to introduce yet another Kconfig option. It complicates the code but brings no benefit.

Reply:
This was requested [earlier](https://github.com/zephyrproject-rtos/zephyr/pull/24884#discussion_r419265245).

Reply:
@mnkp the difference is that the `CONFIG_HWINFO_HAS_DRIVER` can be checked at compile time.

Reply:
Sorry, didn't notice the earlier request. This approach however is not clean. We have and should stay with one driver called hwinfo. With this setup it's not really clear what reset cause really is. It looks like we have two separate drivers now, one called "device id" and second called "reset cause". Even if they are packed in the same implementation file. Looks messy.

We could implement it as an extension of features of hwinfo driver, i.e. a Kconfig option would enable extra features in the hwinfo API. Though I still see no good reason. This extra option doesn't allow to write portable applications since the application (assuming the same code would run on different SoC families) would still need to call `hwinfo_get_reset_cause` function at runtime to verify that the reset cause it requires is supported by the hardware.

Reply:
@mnkp when we introduced the hwinfo API, we agreed that it should exactly look how it is now. We had discussions in the API meeting and agreed on this approach.
However, the `CONFIG_*_HAS_DRIVER` is a nice to have and IMO we should keep it. Maybe someone wants to have static asserts in his code. The thing is that hwinfo has no device-tree and therefor Kconfig is the only way to her compile-time information about the availability.

Reply:
I may be missing some arguments that were raised earlier. Could we maybe have a short discussion during next API meeting? And if not let's leave it as is.

Reply:
The concerns raised by @mnkp here matches the concerns I raised in this PR (https://github.com/zephyrproject-rtos/zephyr/pull/24884#issuecomment-623309496) a year ago.

Reply:
Whatever your decide after your internal discussion please let me know and I'll make the changes appropriately.